src/issuance.rs

```diff
@@ -733,4 +828,265 @@ mod tests {
+        let prev_finalized = &mut HashSet::new();
+
+        // 1. Try too long description
+        signed
+            .actions
+            .first_mut()
+            .unwrap()
+            .modify_descr(String::from_utf8(vec![b'X'; 513]).unwrap());
+        let finalized = verify_issue_bundle(&signed, sighash, prev_finalized);
+
+        assert_eq!(finalized.unwrap_err(), WrongAssetDescSize);
+
+        // 2. Try empty description
+        signed
+            .actions
+            .first_mut()
+            .unwrap()
+            .modify_descr("".to_string());
+        let finalized = verify_issue_bundle(&signed, sighash, prev_finalized);
+
+        assert_eq!(finalized.unwrap_err(), WrongAssetDescSize);
+    }
+}
+
+/// Generators for property testing.
+#[cfg(any(test, feature = "test-dependencies"))]
+#[cfg_attr(docsrs, doc(cfg(feature = "test-dependencies")))]
+pub mod testing {
+    use crate::issuance::{IssueAction, IssueBundle, Prepared, Signed, Unauthorized};
+    use crate::keys::{IssuerAuthorizingKey, IssuerValidatingKey, SpendingKey};
+    use crate::note::testing::arb_zsa_note;
+    use crate::value::testing::arb_note_value_bounded;
+    use crate::value::{NoteValue, MAX_NOTE_VALUE};
+    use proptest::collection::vec;
+    use proptest::prelude::*;
+    use proptest::prop_compose;
+    use proptest::string::string_regex;
+    use rand::{rngs::StdRng, SeedableRng};
+
+    prop_compose! {
+        /// Generate a uniformly distributed RedDSA issuer authorizing key.
+        pub fn arb_issuer_authorizing_key()(rng_seed in prop::array::uniform32(prop::num::u8::ANY)) -> IssuerAuthorizingKey {
+            let mut rng = StdRng::from_seed(rng_seed);
+            IssuerAuthorizingKey::from(&SpendingKey::random(&mut rng))
+        }
+    }
+
+    prop_compose! {
+        /// Generate a uniformly distributed RedDSA issuer validating key.
+        pub fn arb_issuer_validating_key()(isk in arb_issuer_authorizing_key()) -> IssuerValidatingKey {
+            IssuerValidatingKey::from(&isk)
+        }
+    }
+
+    prop_compose! {
+        /// Generate an issue action given note value
+        pub fn arb_issue_action(value: NoteValue)(
+            note in arb_zsa_note(value),
+            asset_descr in string_regex(".{1,512}").unwrap()
+        ) -> IssueAction {
+            IssueAction::new(asset_descr, &note)
+        }
+    }
+
+    /// Generate an issue action having issued values less than MAX_NOTE_VALUE / n_actions.
+    pub fn arb_issue_action_n(n_actions: usize) -> impl Strategy<Value = IssueAction> {
+        let value_gen = Strategy::boxed(arb_note_value_bounded(MAX_NOTE_VALUE / n_actions as u64));
+        value_gen.prop_flat_map(arb_issue_action)
+    }
+
+    prop_compose! {
+        /// Generate an arbitrary issue bundle with fake authorization data. This bundle does not
+        /// necessarily respect consensus rules; for that use
+        /// TODO [`crate::builder::testing::arb_issue_bundle`]
```

Comment:
Repeating the same design they had - current proptests create invalid random data structures, while realistic ones are part of e2e and builder.rs. Since we have separate PRs for units/props and e2e - I decided to work on realistic data structures in e2e context